source/tests/utilities/test_helpers.h

```diff
@@ -1,21 +1,74 @@
 #define TEST_HELPERS
 
-#include <gtest/gtest.h>  // For EXPECT matchers
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <optional>
 
 #include "client_helpers.h"
 
-inline void expect_driver_error(const nidevice_grpc::experimental::client::grpc_driver_error& ex, int32_t expected_error)
+#define EXPECT_THROW_DRIVER_ERROR(statement_, expected_error_)                 \
+  EXPECT_THROW({                                                               \
+    try {                                                                      \
+      statement_;                                                              \
+    }                                                                          \
+    catch (const nidevice_grpc::experimental::client::grpc_driver_error& ex) { \
+      EXPECT_DRIVER_ERROR(ex, expected_error_);                                \
+      throw;                                                                   \
+    }                                                                          \
+  }, nidevice_grpc::experimental::client::grpc_driver_error)
+
+#define EXPECT_THROW_DRIVER_ERROR_WITH_SUBSTR(statement_, expected_error_, message_substring_) \
+  EXPECT_THROW({                                                                               \
+    try {                                                                                      \
+      statement_;                                                                              \
+    }                                                                                          \
+    catch (const nidevice_grpc::experimental::client::grpc_driver_error& ex) {                 \
+      EXPECT_DRIVER_ERROR_WITH_SUBSTR(ex, expected_error_, message_substring_);                \
+      throw;                                                                                   \
+    }                                                                                          \
+  }, nidevice_grpc::experimental::client::grpc_driver_error)
+
+#define EXPECT_DRIVER_ERROR(exception_, expected_error_)             \
+  do {                                                               \
+    EXPECT_EQ(::grpc::StatusCode::UNKNOWN, exception_.StatusCode()); \
+    EXPECT_LT(expected_error_, 0);                                   \
+    auto driver_error = get_driver_error(exception_);                \
+    EXPECT_THAT(driver_error, ::testing::Optional(expected_error_)); \
+  } while(false)
+
+#define EXPECT_DRIVER_ERROR_WITH_SUBSTR(exception_, expected_error_, message_substring_) \
+  do {                                                                                   \
+    EXPECT_DRIVER_ERROR(exception_, expected_error_);                                    \
+    EXPECT_THAT(exception_.what(), ::testing::HasSubstr(message_substring_));            \
+  } while(false)
+
+inline std::optional<int> get_driver_error(const std::multimap<std::string, std::string>& metadata)
```

Comment:
Hopefully I'm wrong, but this might not work on the linux build. I couldn't get variant to work (granted [quite a while ago](https://github.com/ni/grpc-device/pull/326#discussion_r695869290)).

@reckenro has the linux toolchain been updated or is there any talk of doing so? We we sort of "stuck" because RT doesn't (didn't?) publish a toolchain in the same format we're using anymore. I know there's a separate bitbake implementation to add `grpc-device` to the feeds, but I'm not sure how to adapt that to run pre-submit (and possibly to publish binaries on grpc-device releases).

Reply:
I did a test build on a Linux machine running Debian 11, but yeah, I see the Linux RT build is failing:
```
In file included from /home/runner/work/grpc-device/grpc-device/source/tests/system/nidaqmx_driver_api_tests.cpp:14:0:
/home/runner/work/grpc-device/grpc-device/./source/tests/utilities/test_helpers.h:47:13: error: ‘optional’ in namespace ‘std’ does not name a template type
 inline std::optional<int> get_driver_error(const std::multimap<std::string, std::string>& metadata)
             ^~~~~~~~
/home/runner/work/grpc-device/grpc-device/./source/tests/utilities/test_helpers.h:57:13: error: ‘optional’ in namespace ‘std’ does not name a template type
 inline std::optional<int> get_driver_error(const std::multimap<grpc::string_ref, grpc::string_ref>& metadata)
             ^~~~~~~~
/home/runner/work/grpc-device/grpc-device/./source/tests/utilities/test_helpers.h:69:13: error: ‘optional’ in namespace ‘std’ does not name a template type
 inline std::optional<int> get_driver_error(const nidevice_grpc::experimental::client::grpc_driver_error& ex)
             ^~~~~~~~
```

Reply:
Instead of std::optional, I'll use `ADD_FAILURE() << message` and return 0.